### PR TITLE
UCT/IB/DC: add DC compatibility check

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
@brminich @yosefe

### CODE REVIEW

File: src/uct/ib/dc/base/dc_iface.c [hunk 1]
```diff
@@ -219,11 +219,27 @@ void uct_dc_iface_set_quota(uct_dc_iface_t *iface, uct_dc_iface_config_t *config
                                 ucs_min(iface->super.config.tx_qp_len, config->quota);
 }
 
+static uint8_t uct_dc_version(uct_ib_device_t *dev)
```
Comment (yosefe) [inline src/uct/ib/dc/base/dc_iface.c]: IMHO better rename to uct_ib_device_get_dc_version(), and move to ib_device.c

File: src/uct/ib/dc/base/dc_iface.c [hunk 2]
```diff
@@ -323,14 +342,40 @@ ucs_status_t uct_dc_iface_query(uct_dc_iface_t *iface,
...
+                                                        uct_dc_iface_t);
+    uct_dc_iface_addr_t *addr = (uct_dc_iface_addr_t *)iface_addr;
+
+    if ((iface_addr  != NULL) &&
+        (UCT_DC_IFACE_ADDR_TM_ENABLED(addr) !=
+         UCT_RC_IFACE_TM_ENABLED(&iface->super))) {
+        return 0;
+    }
+
+    if (!(addr->flags & iface->version_flag)) {
```
Comment (yosefe) [inline src/uct/ib/dc/base/dc_iface.c]: maybe this condition can be part of previous 'if' ?

File: src/uct/ib/dc/base/dc_iface.h [hunk 3]
```diff
@@ -15,14 +15,25 @@
 
 #define UCT_DC_IFACE_MAX_DCIS   16
 
+#define UCT_DC_IFACE_ADDR_TM_ENABLED(_addr) \
+    (!!(_addr->flags & UCT_DC_IFACE_ADDR_HW_TM))
```
Comment (yosefe) [inline src/uct/ib/dc/base/dc_iface.h]: need to put () around _addr

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6057/ for details (Mellanox internal link).

File: src/uct/ib/dc/base/dc_iface.c (hunk 1 above)
Comment (evgeny-leksikov) [reply src/uct/ib/dc/base/dc_iface.c]: it would cause dependency of base module from dc, reworked a bit different way

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6065/ for details (Mellanox internal link).

**Pull request merged**